lntest/itest/lnd_zero_conf_test.go

```diff
@@ -286,110 +230,89 @@ func testOptionScidAlias(net *lntest.NetworkHarness, t *harnessTest) {
+	dave := ht.NewNode("Dave", scidAliasArgs)
 
 	// Ensure Bob, Carol are connected.
-	net.EnsureConnected(t.t, net.Bob, carol)
+	ht.EnsureConnected(bob, carol)
 
 	// Ensure Carol, Dave are connected.
-	net.EnsureConnected(t.t, carol, dave)
+	ht.EnsureConnected(carol, dave)
 
 	// Give Carol some coins so she can open the channel.
-	net.SendCoins(t.t, btcutil.SatoshiPerBitcoin, carol)
+	ht.FundCoins(btcutil.SatoshiPerBitcoin, carol)
 
 	chanAmt := btcutil.Amount(1_000_000)
 
-	params := lntest.OpenChannelParams{
+	params := lntemp.OpenChannelParams{
 		Amt:            chanAmt,
 		Private:        private,
 		CommitmentType: lnrpc.CommitmentType_ANCHORS,
 		ScidAlias:      chantype,
 	}
-	fundingPoint := openChannelAndAssert(t, net, carol, dave, params)
+	fundingPoint := ht.OpenChannel(carol, dave, params)
 
+	// Make sure Bob knows this channel if it's public.
 	if !private {
-		err := net.Bob.WaitForNetworkChannelOpen(fundingPoint)
-		require.NoError(t.t, err, "bob didn't report channel")
+		ht.AssertTopologyChannelOpen(bob, fundingPoint)
 	}
 
-	err := carol.WaitForNetworkChannelOpen(fundingPoint)
-	require.NoError(t.t, err, "carol didn't report channel")
-	err = dave.WaitForNetworkChannelOpen(fundingPoint)
-	require.NoError(t.t, err, "dave didn't report channel")
-
 	// Assert that a payment from Carol to Dave works as expected.
 	daveInvoiceParams := &lnrpc.Invoice{
 		Value:   int64(10_000),
 		Private: true,
 	}
-	daveInvoiceResp, err := dave.AddInvoice(ctxb, daveInvoiceParams)
-	require.NoError(t.t, err)
-	_ = sendAndAssertSuccess(
-		t, carol, &routerrpc.SendPaymentRequest{
-			PaymentRequest: daveInvoiceResp.PaymentRequest,
-			TimeoutSeconds: 60,
-			FeeLimitMsat:   noFeeLimitMsat,
-		},
+	daveInvoiceResp := dave.RPC.AddInvoice(daveInvoiceParams)
+	ht.CompletePaymentRequests(
+		carol, []string{daveInvoiceResp.PaymentRequest},
 	)
 
 	// We'll now open a regular public channel between Bob and Carol and
 	// assert that Bob can pay Dave. We'll also assert that the invoice
 	// Dave issues has the startingAlias as a hop hint.
-	fundingPoint2 := openChannelAndAssert(
-		t, net, net.Bob, carol,
-		lntest.OpenChannelParams{
-			Amt: chanAmt,
-		},
-	)
-
-	err = net.Bob.WaitForNetworkChannelOpen(fundingPoint2)
-	require.NoError(t.t, err)
-	err = carol.WaitForNetworkChannelOpen(fundingPoint2)
-	require.NoError(t.t, err)
+	p := lntemp.OpenChannelParams{
+		Amt: chanAmt,
+	}
+	fundingPoint2 := ht.OpenChannel(bob, carol, p)
+
+	defer func() {
+		// TODO(yy): remove the sleep once the following bug is fixed.
+		// When the payment is reported as settled by Bob, it's
+		// expected the commitment dance is finished and all subsequent
+		// states have been updated. Yet we'd receive the error `cannot
+		// co-op close channel with active htlcs` or `link failed to
+		// shutdown` if we close the channel. We need to investigate
+		// the order of settling the payments and updating commitments
+		// to understand and fix.
+		time.Sleep(2 * time.Second)
```

Comment:
Could there be an alternative to `Sleep`, perhaps waiting for some state change with `Eventually` ?

Reply:
See [previous discussion](https://github.com/lightningnetwork/lnd/pull/6776#discussion_r957880737). In short, we want to expose it here so we don't forget to fix it.